src/storage/src/source/postgres.rs

```diff
@@ -542,24 +565,29 @@ impl PostgresTaskInfo {
         }
         self.metrics.lsn.set(self.lsn.into());
         client.simple_query("COMMIT;").await?;
+
+        // close the current `row_sender` context after we are sure we are not erroring out.
+        // Otherwise, `revert_snapshot` will close it
+        self.row_sender.close_lsn(self.lsn).await;
         Ok(())
     }
 
     /// Reverts a failed snapshot by deleting any processed rows from the dataflow.
     async fn revert_snapshot<R: Read + Seek>(
-        snapshot_tx: &mut Transaction,
+        &mut self,
         mut reader: R,
     ) -> Result<(), anyhow::Error> {
-        tokio::task::block_in_place(|| -> Result<(), anyhow::Error> {
-            let len = reader.seek(SeekFrom::Current(0))?;
-            reader.seek(SeekFrom::Start(0))?;
-            let mut reader = reader.take(len);
-            while reader.limit() > 0 {
-                let row = bincode::deserialize_from(&mut reader)?;
-                snapshot_tx.delete(row);
-            }
-            Ok(())
-        })
+        let len = reader.seek(SeekFrom::Current(0))?;
+        reader.seek(SeekFrom::Start(0))?;
+        let mut reader = reader.take(len);
+        // TODO(guswynn): this is blocking code on a tokio task, but itll soon be gone:
+        // https://github.com/MaterializeInc/materialize/pull/13159/files
```

Comment:
I assume this is already gone on main, right?

Reply:
yes!